api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ConceptTranslatorImpl.java

```diff
@@ -48,45 +57,22 @@ public class ConceptTranslatorImpl implements ConceptTranslator {
 	
 	@Override
 	public CodeableConcept toFhirResource(@Nonnull Concept concept) {
-		if (concept == null) {
-			return null;
-		}
+		return toFhirResources(Collections.singletonList(concept)).get(0);
+	}
+	
+	@Override
+	public List<CodeableConcept> toFhirResources(Collection<Concept> openmrsConcepts) {
+		ConceptTranslatorContext context = getConceptTranslatorContext();
 		
-		CodeableConcept codeableConcept = new CodeableConcept();
-		codeableConcept.setText(concept.getDisplayString());
-		addConceptCoding(codeableConcept.addCoding(), null, concept.getUuid(), concept);
-		//map of <systemUrl ,<mapType , code>> ie { "http://loinc.org” : { "SAME-AS" : "108-5", "NARROWER-THAN": "108-8" }}
-		Map<String, Map<String, String>> systemUrlToCodeMap = new HashMap<>();
-		for (ConceptMap mapping : concept.getConceptMappings()) {
-			if (mapping.getConceptMapType() != null) {
-				ConceptMapType mapType = mapping.getConceptMapType();
-				boolean sameAs = mapType.getUuid() != null && mapType.getUuid().equals(ConceptMapType.SAME_AS_MAP_TYPE_UUID);
-				sameAs = sameAs || (mapType.getName() != null && mapType.getName().equalsIgnoreCase("SAME-AS"));
-				ConceptReferenceTerm crt = mapping.getConceptReferenceTerm();
-				String sourceUrl = conceptSourceService.getUrlForConceptSource(crt.getConceptSource());
-				if (sourceUrl != null) {
-					if (sameAs) {
-						addSystemToCodeMap(systemUrlToCodeMap, sourceUrl, "SAME-AS", crt.getCode());
-					} else {
-						addSystemToCodeMap(systemUrlToCodeMap, sourceUrl, mapType.getName(), crt.getCode());
-					}
-				}
-			}
-		}
+		return openmrsConcepts.stream().map(concept -> toFhirResource(concept, context)).collect(Collectors.toList());
+	}
+	
+	@Override
+	public Map<Concept, CodeableConcept> toFhirResourcesMap(Collection<Concept> openmrsConcepts) {
+		ConceptTranslatorContext context = getConceptTranslatorContext();
```

Comment:
I think you need to call it via self.getConceptTranslatorContext() and add 
```
@Setter(value = PROTECTED, onMethod_ = @Autowired)
private ConceptTranslator self;
```
For the method to be called via proxy. It's explained at https://www.baeldung.com/spring-invoke-cacheable-other-method-same-bean

Reply:
Please see my comment above about not introducing this toFhirResourcesMap at all.